TKCurrencyTextField/Classes/TKCurrencyTextField.swift

```diff
@@ -8,38 +8,76 @@
 
 import UIKit
 
+public typealias ReplacementString = String
+public typealias ReplacementStringHandler = (UITextField, NSRange, ReplacementString) -> () // Can be used on textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool
+
 @IBDesignable open class TKCurrencyTextField: UITextField {
     
     // MARK:- Public
     
     @IBInspectable public var maxDigits: Int = 15
-    @IBInspectable public var defaultValue: Double = 0.00
-    @IBInspectable public var currencySymbol: Bool = true
-    public var locale: Locale = Locale.current {
+    @IBInspectable public var defaultValue: Double = 0.00 {
         didSet {
-            setAmount(defaultValue)
+            self.textDidChange()
         }
     }
-    
-    public func setAmount (_ amount: Double) {
-        let textFieldStringValue = amount.currencyStringValue(with: locale, currencySymbol)
-        text = textFieldStringValue
+    @IBInspectable public var currencySymbol: Bool = true {
+        didSet {
+            self.textDidChange()
+        }
     }
-    
-    public var getAmount: Double {
-        if let text = text {
-            return text.currencyStringToDouble(with: locale)
+    public var locale: Locale = Locale.current {
+        didSet {
+            self.textDidChange()
         }
-        return defaultValue
     }
     
-    // MARK:- Private vars
-    
-    private var _maskHandler: ((_ textField: UITextField, _ range: NSRange, _ replacementString: String) -> ())? = nil
-    var maskHandler: ((_ textField: UITextField, _ range: NSRange, _ replacementString: String) -> ())? {
+    public var amount: Double {
         get {
-            return _maskHandler
+            if let text = text {
+                return text.currencyStringToDouble(with: locale)
+            }
+            
+            return defaultValue
+        }
+        
+        set {
+            let textFieldStringValue = amount.currencyStringValue(with: locale, currencySymbol)
+            text = textFieldStringValue
+        }
+    }
+
+    lazy var maskHandler: ReplacementStringHandler = {
+        (textField, range, replacementString) in
+        let nsString: NSString = textField.text as NSString? ?? ""
```

Comment:
@damboscolo 
Please, check if optional character (?) should be after `NSString` or after `as`.
I belive this simbol should be after `as` like:

```swift
let nsString: NSString = textField.text as? NSString ?? ""
```